packages/ui/src/components/va-tabs/hooks/useTabsView.ts

```diff
@@ -0,0 +1,414 @@
+  const wrapper = ref<HTMLElement | null>(null)
+  const container = ref<HTMLElement | null>(null)
+  const tabs = ref<HTMLElement | null>(null)
+
+  const sliderHeight = ref<number | null>(null)
+  const sliderWidth = ref<number | null>(null)
+  const sliderOffsetX = ref(0)
+  const sliderOffsetY = ref(0)
+  const showPagination = ref(false)
+  const tabsContentOffset = ref(0)
+  const resizeObserver = ref<ResizeObserver | null>(null)
+  const startingXPoint = ref(0)
+  const animationIncluded = ref(false)
+  const { colorComputed } = useColor(props)
+  const { valueComputed } = useStateful(props, emit)
+  const tabsService = ref<any>()
+  const tabConfig = reactive({
+    VaTab: {
+      color: props.color,
+    },
+  })
+
+  const computedClass = computed(() => {
+    const {
+      left,
+      right,
+      center,
+      grow,
+      disabled,
+    } = props
+
+    return {
+      'va-tabs__container--left': left && !right && !center && !grow,
+      'va-tabs__container--right': right,
+      'va-tabs__container--center': center,
+      'va-tabs__container--grow': grow,
+      'va-tabs__container--disabled': disabled,
+    }
+  })
+
+  const computedTabsClass = computed(() => ({ 'va-tabs--vertical': props.vertical }))
+
+  const tabSelected = computed(() => valueComputed)
+
+  const sliderStyles = computed(() => {
+    if (props.hideSlider) {
+      return { display: 'none' }
+    }
+
+    const transition = animationIncluded.value ? 'var(--va-tabs-slider-wrapper-transition)' : ''
+
+    if (props.vertical) {
+      return {
+        'background-color': colorComputed.value,
+        height: `${sliderHeight.value}px`,
+        transform: `translateY(-${sliderOffsetY.value}px) translateX(${sliderOffsetX.value}px)`,
+        transition,
+      }
+    }
+
+    return {
+      'background-color': colorComputed.value,
+      width: `${sliderWidth.value}px`,
+      transform: `translateY(-${sliderOffsetY.value}px) translateX(${sliderOffsetX.value}px)`,
+      transition,
+    }
+  })
+
+  const paginationControlledStyles = computed(() => {
+    // Prevents the movement of vertical tabs
+    if (props.vertical) {
+      return {
+        transform: 'translateX(0px)',
+      }
+    }
+
+    const transition = animationIncluded.value ? 'transform ease 0.3s' : ''
+
+    return {
+      transform: `translateX(${startingXPoint.value - tabsContentOffset.value}px)`,
+      transition,
+    }
+  })
+
+  const disablePaginationLeft = computed(() => tabsContentOffset.value === 0)
+
+  const disablePaginationRight = computed(() => {
+    const lastTab = tabsService.value?.tabs[tabsService.value.tabs.length - 1]
+    const containerClientWidth = getClientWidth(container.value)
+
+    return (lastTab?.rightSidePosition || 0) <= tabsContentOffset.value + containerClientWidth || (lastTab?.leftSidePosition || 0) <= tabsContentOffset.value
+  })
+
+  // Methods
```

Comment:
One of the purposes of composition api is to group entities by their logical concerns rather than type of entity
https://vuejs.org/guide/extras/composition-api-faq.html#more-flexible-code-organization

Reply:
After a few days working with it, I understood it. :)